components/feature/tabs/src/main/java/mozilla/components/feature/tabs/toolbar/TabsToolbarFeature.kt

```diff
@@ -14,13 +15,23 @@ import mozilla.components.concept.toolbar.Toolbar
 class TabsToolbarFeature(
     toolbar: Toolbar,
     sessionManager: SessionManager,
+    sessionId: String? = null,
     showTabs: () -> Unit
 ) {
     init {
-        val tabsAction = TabCounterToolbarButton(
-            sessionManager,
-            showTabs)
+        run {
+            sessionManager.runWithSessionId(sessionId) {
+                if (it.isCustomTabSession()) {
+                    return@run
+                }
+            }
 
-        toolbar.addBrowserAction(tabsAction)
+            val tabsAction = TabCounterToolbarButton(
+                sessionManager,
+                showTabs
+            )
+
+            toolbar.addBrowserAction(tabsAction)
+        }
```

Comment:
Why is this wrapped in a `run` block?

Reply:
I was adding an early return if it's a custom tab session, but you can't "return" in an init block, so [some recommendations](https://stackoverflow.com/questions/46244409/why-is-return-not-allowed-in-kotlin-init-block) were to wrap it in a run block instead and return that.

Reply:
So early returns _are_ hard to read, you say? ;)

That's fair though. How about..

```suggestion
    init {
        sessionManager.runWithSession(sessionId) {
            if (!it.isCustomTabSession()) {
                return@runWithSession
            }
           
            val tabsAction = TabCounterToolbarButton(
                sessionManager,
                showTabs
            )

            toolbar.addBrowserAction(tabsAction)
        }
    }
```